### PR TITLE
Add organization using PUDL section to readme

### PR DESCRIPTION
**Pull request by bendnorman**
# Overview

Someone as CSV conf recommended we add a section to our README/docs where orgs using PUDL data can add themselves. I think this is an easy way to find more of our users and build trust with prospective users.

I once added PUDL to pydantic's readme but the section as removed with v2. 

**Should we structure this information so that it can live in our docs and README?**

# Testing

How did you make sure this worked? How can a reviewer verify this?

```[tasklist]
# To-do list
- [ ] If updating analyses or data processing functions: make sure to update or write data validation tests (e.g.,  `test_minmax_rows()`)
- [ ] Update the [release notes](../docs/release_notes.rst): reference the PR and related issues.
- [ ] Ensure docs build, unit & integration tests, and test coverage pass locally with `make pytest-coverage` (otherwise the merge queue may reject your PR)
- [ ] Review the PR yourself and call out any questions or issues you have
- [ ] For minor ETL changes or data additions, once `make pytest-coverage` passes, make sure you have a fresh full PUDL DB downloaded locally, materialize new/changed assets and all their downstream assets and [run relevant data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation) using `pytest` and `--live-dbs`.
- [ ] For significant ETL, data coverage or analysis changes, once `make pytest-coverage` passes, ensure the full ETL runs locally and [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation) using `make pytest-validate` (a ~10 hour run). If you can't run this locally, run the `build-deploy-pudl` GitHub Action (or ask someone with permissions to). Then, check the logs on the `#pudl-deployments` Slack channel or `gs://builds.catalyst.coop`.
```

### CODE REVIEW

**Comment by bendnorman** [comment]
It'd also be nice to add a couple of orgs to the list before we merge this in.

**Comment by e-belfer** [comment]
@bendnorman Do you want to add organizations we already know are using PUDL data to this PR? Possibly we can tag some of our known downstream users on Github and just get a confirmation that they accept being added, so we don't start with an empty list?

File: README.rst [hunk 1]
```diff
@@ -159,6 +159,15 @@ summary:
 * `The PUDL Development Environment <https://catalystcoop-pudl.readthedocs.io/en/nightly/dev/dev_setup.html>`__
   lets you run the PUDL data processing pipeline locally.
 
+Organizations using PUDL
+------------------------
+
+This is an imcomplete list of organizations that have used PUDL in their work.
```
Comment (e-belfer) [inline README.rst:165]: Tiny typo - Incomplete?

**Comment by e-belfer** [comment]
By my count, that would be:
- The [Utility Transition Hub](https://utilitytransitionhub.rmi.org/) and [Optimus](https://utilitytransitionhub.rmi.org/optimus/) financial modelling tool at RMI (@arengel)
- [PowerGenome](https://github.com/PowerGenome/PowerGenome) at ZEROLab at Princeton University (@gschivley)
- [Open Grid Emissions](https://singularity.energy/open-grid-emissions) at Singularity Energy (@grgmiller)
- [PyPSA - USA](https://github.com/PyPSA/pypsa-usa) (@ktehranchi)

If I've just tagged you: we're taking the long overdue step of adding organizations that use our data to our README. Just wanted to confirm that you all are OK with being listed here?

**Comment by bendnorman** [comment]
@e-belfer do you think the list should include a description of how the project is using PUDL or just a link to the org?

**Comment by e-belfer** [comment]
> @e-belfer do you think the list should include a description of how the project is using PUDL or just a link to the org?

I think the project and organization name with links should be enough here, personally.

**Review by e-belfer — APPROVED** [review]
Thanks for taking initiative here!

Non-blocking: Think it'd be great to also add this same list on the home page (`index.rst`) of our readthedocs, but this is a good start.

**Comment by zaneselvans** [comment]
Some others:
* Grid Strategies, LLC in [this report](https://gridstrategiesllc.com/wp-content/uploads/2023/12/National-Load-Growth-Report-2023.pdf) which everyone is freaking out about.
* Invenia Labs (it's been a while)
* Energy Innovation, LLC (Coal Cost Crossover, day-to-day calculations)
* Carbon Tracker
* Institute for Policy Integrity at NYU Law School

**Comment by bendnorman** [comment]
@e-belfer I just made that change!

@zaneselvans, I'll reach out to those other organizations to confirm they're okay with us including them.

**Pull request merged**